### DISCUSSION TITLE
Registering JSRuntime in the browser version of AvaloniaUI

### DISCUSSION THREAD

**Discussion by IVANKIRICHENKO04** [AvaloniaUI/Avalonia#16323] [General]
I'm trying to import JS methods into .Net and vice versa, at the moment I managed to run a .net method from js, there was a problem with calling a js method from .net, how to correctly register JSRuntime when compiling an application, this is what I have implemented now
```c#
[assembly: SupportedOSPlatform("browser")]

internal partial class Program
{
    private static async Task Main(string[] args)
    {
        var serviceProvider = new ServiceCollection()
            .AddScoped<IJSRuntime, WebAssemblyJSRuntime>()
            //.AddSingleton<IJSUnmarshalledRuntime, WebAssemblyJSUnmarshalledRuntime>() // Добавляем поддержку незамаршаленных вызовов
            .BuildServiceProvider();

        var jsRuntime = serviceProvider.GetRequiredService<IJSRuntime>();
        JSRuntimeProvider.Initialize(jsRuntime);

        await BuildAvaloniaApp()
            .WithInterFont()
            .UseReactiveUI()
            .StartBrowserAppAsync("out");
    }

    public static AppBuilder BuildAvaloniaApp()
        => AppBuilder.Configure<App>();
}
```

**Comment by maxkatz6** [comment]
`IJSRuntime` interop is not supported in .NET, it's specifically Blazor feature.

If you want an universal interop JS model that works with any .NET WASM app (including Avalonia), use JSImport and JSExport - https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop?view=aspnetcore-8.0
